cme/modules/enum_av.py

```diff
@@ -3,106 +3,102 @@
-
+        
     def options(self, context, module_options):
-        """ """
+        """Placeholder for module options."""
         pass
 
     def on_login(self, context, connection):
-        success = 0
+        target = self._get_target(connection)
+        context.log.debug(f"Detecting installed services on {target} using LsarLookupNames()...")
+
+        results = self._detect_installed_services(context, connection, target)
+        self._detect_running_processes(context, connection, results)
+
+        self.dump_results(results, connection.hostname, context)
+
+    def _get_target(self, connection):
+        return connection.host if not connection.kerberos else f"{connection.hostname}.{connection.domain}"
+
+    def _detect_installed_services(self, context, connection, target):
         results = {}
-        target = connection.host if not connection.kerberos else connection.hostname + "." + connection.domain
-        context.log.debug("Detecting installed services on {} using LsarLookupNames()...".format(target))
 
         try:
             lsa = LsaLookupNames(
-                connection.domain,
-                connection.username,
-                connection.password,
-                target,
-                connection.kerberos,
-                connection.domain,
-                connection.lmhash,
-                connection.nthash,
-                connection.aesKey,
+                domain=connection.domain,
+                username=connection.username,
+                password=connection.password,
+                remote_name=target,
+                k=connection.kerberos,
+                kdcHost=connection.domain,
+                lmhash=connection.lmhash,
+                nthash=connection.nthash,
+                aesKey=connection.aesKey
             )
-            dce, rpctransport = lsa.connect()
+
+            dce, _ = lsa.connect()
             policyHandle = lsa.open_policy(dce)
 
-            for i, product in enumerate(conf["products"]):
+            for product in conf["products"]:
                 for service in product["services"]:
                     try:
                         lsa.LsarLookupNames(dce, policyHandle, service["name"])
                         context.log.info(f"Detected installed service on {connection.host}: {product['name']} {service['description']}")
-                        if product["name"] not in results:
-                            results[product["name"]] = {"services": []}
-                        results[product["name"]]["services"].append(service)
-                    except Exception as e:
+                        results.setdefault(product["name"], {"services": []})["services"].append(service)
+                    except:
                         pass
-            success += 1
+
         except Exception as e:
             context.log.fail(str(e))
 
+        return results
+
+    def _detect_running_processes(self, context, connection, results):
```

Comment:
Why is this tagged as a private method (underscore)?

Reply:
encapsulation for future use :)